torch/fx/passes/_tensorify_python_scalars.py

```diff
@@ -0,0 +1,262 @@
+from __future__ import annotations
+
+import logging
+from typing import List, Union
+
+import torch
+import torch.fx as fx
+from torch._prims_common import get_computation_dtype
+from torch._subclasses import fake_tensor  # noqa: TCH001
+from torch.fx._utils import lazy_format_graph_code
+from torch.fx.experimental.symbolic_shapes import ShapeEnv  # noqa: TCH001
+from torch.fx.graph_module import GraphModule  # noqa: TCH001
+
+# TODO: refactor
+from torch.fx.passes.runtime_assert import _get_sym_val
+from torch.fx.proxy import MetaProxy
+from torch.utils._sympy.reference import TensorReferenceAnalysis
+
+
+__all__: List[str] = []
+
+log = logging.getLogger(__name__)
+graph_code_log = torch._logging.getArtifactLogger(__name__, "graph_code")
+
+# The general shape of this transformation is to look for Tensor operations
```

Comment:
I think we should add an instruction count microbenchmark for this case, to be able to understand the effect of this conversion on compile time.

Reply:
This is a bit difficult since without this pass, we can't even finish compiling. Inductor will choke on the .item() calls.

Reply:
I meant a micro benchmark where we compile a function with float inputs. does this PR will always convert all float input to tensors?

Reply:
This one doesn't but the following PR does: https://github.com/pytorch/pytorch/pull/125325